reward/staking_manager.go

```diff
@@ -132,12 +126,10 @@ func (sm *StakingManager) handleChainHeadEvent() {
 		// Handle ChainHeadEvent
 		case ev := <-sm.chainHeadChan:
 			if sm.governanceHelper.ProposerPolicy() == params.WeightedRandom {
-				stakingBlockNum := ev.Block.NumberU64() - ev.Block.NumberU64()%sm.governanceHelper.StakingUpdateInterval()
-				if cachedStakingInfo := sm.stakingInfoCache.get(stakingBlockNum); cachedStakingInfo == nil {
-					stakingInfo, _ := sm.updateStakingInfo(stakingBlockNum)
-					if stakingInfo == nil {
-						logger.Error("Failed to update stakingInfoCache", "blockNumber", ev.Block.NumberU64(), "stakingNumber", stakingBlockNum)
-					}
+				// check and update if staking info is not valid beforehand
+				stakingManager := sm.GetStakingInfo(ev.Block.NumberU64())
+				if stakingManager == nil {
+					logger.Error("unable to fetch staking info")
```

Comment:
Please, add block number info on the log.
```suggestion
					logger.Error("unable to fetch staking info", "blockNumber", ev.Block.NumberU64(), "stakingNumber", stakingBlockNum)
```

Reply:
review applied